lambdas/service/app.py

```diff
@@ -1691,6 +1699,65 @@ def file_url(file_uuid: str, fetch: bool = True, **params: str):
     return f'{url}?{params}'
 
 
+@app.route('/fetch/repo/files/{file_uuid}', methods=['GET'], cors=True, method_spec={
+    'tags': ['TDR'],
+    'parameters': [
+        catalog_param_spec,
+        *file_fqid_parameters_spec,
+    ],
+    'summary': 'Request a download link to a file',
+    'responses': {
+        '200': {
+            'description': format_description('''
+                Emulates the response code and headers of HTTP while bypassing the default
+                client behavior. Note that the actual HTTP response will have status 200 while
+                the `Status` field of the body will be 302.
+
+                The response described here is intended to be processed by client-side
+                Javascript such that the authorization header can be handled in Javascript
+                rather than relying on the native implementation by the web browser.
+            '''),
+            **responses.json_content(
+                schema.object(
+                    Status=int,
+                    Location=str,
+                    Headers=str,
+                )
+            )
+        }
+    }
+})
+def fetch_repo_files(file_uuid):
+    """
+    Request a file for download from the TDR
+
+    :return: A 200 response with a JSON body describing the status of the request.
+    """
+    query_params = app.current_request.query_params or {}
+    validate_params(query_params,
+                    version=str,
+                    catalog=IndexName.validate_catalog_name)
+    require(app.catalog in config.catalogs,
+            f"Invalid catalog '{app.catalog}'.")
+    require(config.catalogs[app.catalog]['repository'] == 'tdr',
+            f"Catalog '{app.catalog}' does not use the TDR repository.")
+    file_version = query_params.get('version')
+    service = FileService(app.catalog, file_uuid, file_version)
```

Comment:
In the long term, services will be long-lived and should not be parametrized with request data.

Let's discuss after stand-up.

Note to self: extract field not warranted since only one work horse method and one helper method. FOr service+helper pattern see manifest service + manifest generator.